### DISCUSSION TITLE
apt-offline not working after synced

### DISCUSSION THREAD

**Discussion by semaka** [rickysarraf/apt-offline#151] [Q&A]
could you please help me to understand how apt-offline is not working with synced packages?
I performed the command below:
> apt-offline install myarchive.zip
lot of packages listed as synced

but after I try to install packages from my archive I have the error below:

> root@KVM:/tmp# apt-get install tigervnc-standalone-server
> Reading package lists... Done
> Building dependency tree
> Reading state information... Done
> E: Unable to locate package tigervnc-standalone-server

I mention that I'm not connected to internet and my sources.list is commented all lines. By the way, on another PC I was able to perform this action without any problem.

Why my package is not seen as soon as I have synced from apt-offline???

thank you,

**Comment by rickysarraf** [comment]
From what you have briefly described, it looks like you downloaded the package but the apt package update indices are not updated. But this is just my guess.

Can you share the sequence of commands and their console outputs, that you ran ?

**Reply by semaka** [reply]
It is exactly what you see above, here is truncated:
>apt-offline install myarchive.zip
tigervnc-standalone-server_1.10.1+dfsg-3_amd64.deb file synced.
ERROR: Path for keyring is invalid: /etc/apt/trusted.gpg
>apt-get install tigervnc-standalone-server
Reading package lists... Done
Building dependency tree
Reading state information... Done
E: Unable to locate package tigervnc-standalone-server

I remember you, same software works without any problem on another Ubuntu 20.04 PC. I think that here could be the problem of repository affecting or something else.

**Reply by rickysarraf** [reply]
I can't ascertain much from this information.

The `--update` command has a significance. As in, it does the main job of updating your apt package indices. Previously, `apt-offline` would run both the `update/upgrade` commands by default. But that caused too many confusions among users about the added round-trip.

So in latest commit this was dropped. We expect the user to pass the options explicitly.
See this commit: https://github.com/rickysarraf/apt-offline/commit/5b8844f2a64e022a3e707c2861dbb6e12b7bccbb

**Reply by rickysarraf** [reply]
We could improve the wiki documentation here, which is now outdated. But the [man page](https://github.com/rickysarraf/apt-offline/blob/master/apt-offline.8) is up-to-date.

**Reply by semaka** [reply]
What it's not clear in my message. I didn't mention how I created myarchive.zip with apt-offline, because that the part done with success without any problem. I mentioned just the part with installing in off-line mode (no upgrade needed) just installing my preferred packages and you have in my example one of it. 
The big question is, why after synced with apt-offline the package can't be located to be installed?

**Reply by semaka** [reply]
So, does anybody knows how apt-get locate the packages synced with apt-offline? This is the biggest question.

**Reply by rickysarraf** [reply]
When run the commands in sequence, as outlined in the apt-offline manpage, it should all be transparent to the user.

I gave this talk, which may help your case.
https://www.youtube.com/watch?v=3Z7s-8vsWsw&t=79s

**Reply by semaka** [reply]
your video advice is the happy case when everything works great. I can do the same with first PC which works like a charm but on the second PC in offline mode same packages not working with the error mentioned. I didn't ask you about how apt-offline to be used, because I succeed with it on one PC. I came to you with exactly a problem reported as Unable to locate package, why this happens if all my packages are synced but I have at the end the error: Path for keyring is invalid: /etc/apt/trusted.grg?

**Reply by rickysarraf** [reply]
That error is when your keyring may not have the file.

* What host system are you on ?
* What version of apt-offline are you using ?
* Have you tried with the latest from the git repository ?

The issue you mentioned is something that was fixed long back. https://github.com/rickysarraf/apt-offline/commit/3787a2b9e4d7ce9d6293dfda596f8d7b23fad81f

**Reply by semaka** [reply]
Do you know for what is used keyring? With or without keyring everything works.
By the way I found the problem. apt-offline works great and extract and copy the needed debs to cache without any problem. 

With deep investigation, understanding that apt-get does not get the debs from cache, I found that, if apt-get has everything commented (as my case) in sources.list it does not go to cache. 

So sources.list should have repositories there but no internet connection and when repositories are not found it goes to cache and install from there. (/var/cache/apt/archives)

Hope this will help other people having my problem too.
helped me the link: 
https://itectec.com/ubuntu/ubuntu-install-a-package-moved-to-var-cache-apt-archives-manually/

**Reply by rickysarraf** [reply]
Please don't fight the system. There's a reason for everything in place.
https://wiki.debian.org/SecureApt

You can workaround all of it but that is NOT a solution.

**Reply by semaka** [reply]
thank you, I'll learn more from this.

**Comment by semaka** [comment]
By the way I found the problem. 
apt-offline works great and extract and copy the needed debs to cache without any problem. 

With deep investigation, understanding that apt-get does not get the debs from cache, I found that, if apt-get has everything commented (as my case) in sources.list it does not go to cache. 

So sources.list should have repositories there but no internet connection and when repositories are not found it goes to cache and install from there. (/var/cache/apt/archives)

Hope this will help other people having my problem too.
helped me the link: 
https://itectec.com/ubuntu/ubuntu-install-a-package-moved-to-var-cache-apt-archives-manually/

**Reply by rickysarraf** [reply]
This is not correct. There's more to it. `apt-offline` and `apt` are supposed to work together. The job of `apt-offline` is to just bridge the gap of network unavailability. Any other change is not needed.

**Reply by semaka** [reply]
yes they work together, but in my problem exposed here I found what could cause an anomaly for not working anymore. It is not normal what happens but I reproduced my problem each time and I saw the correction. Another bug observed by me, is that if sources.list was commented "#" all lines one time, and if apt-get is not working because of this, even if I change back the file with not comment lines the problem is not solved. It is needed a new OS install to work again.

**Reply by rickysarraf** [reply]
Please don't mark it as an answer. It will mislead other readers.

**Reply by semaka** [reply]
but that is the solution why not being as answer???
